packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/DSL_DataSpace_PureProtocolProcessorPlugin.ts

```diff
@@ -385,6 +405,94 @@ export class DSL_DataSpace_PureProtocolProcessorPlugin extends PureProtocolProce
       },
     ];
   }
+
+  V1_getExtraMappingIncludeBuilders(): V1_MappingIncludeBuilder[] {
+    return [
+      (
+        protocol: V1_MappingInclude,
+        parentMapping: Mapping,
+        context: V1_GraphBuilderContext,
+      ): MappingInclude | undefined => {
+        if (protocol instanceof V1_MappingIncludeDataSpace) {
+          const dataSpace = V1_resolveDataSpace(
+            protocol.includedDataSpace,
+            context,
+          );
+          const includedMapping = new MappingIncludeDataSpace(
+            parentMapping,
+            dataSpace.value.defaultExecutionContext.mapping,
+            dataSpace,
+          );
+          return includedMapping;
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  V1_getExtraMappingIncludeTransformers(): V1_MappingIncludeTransformer[] {
+    return [
+      (
+        metamodel: MappingInclude,
+        context: V1_GraphTransformerContext,
+      ): V1_MappingInclude | undefined => {
+        if (metamodel instanceof MappingIncludeDataSpace) {
+          const mappingInclude = new V1_MappingIncludeDataSpace();
+          mappingInclude.includedDataSpace =
+            metamodel.includedDataSpace.valueForSerialization ?? '';
+          return mappingInclude;
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  V1_getExtraMappingIncludeProtocolSerializers(): V1_MappingIncludeProtocolSerializer[] {
+    return [
+      (
+        protocol: V1_MappingInclude,
+      ): PlainObject<V1_MappingInclude> | undefined => {
+        if (protocol instanceof V1_MappingIncludeDataSpace) {
+          return V1_serializeMappingInclude(protocol);
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  V1_getExtraMappingIncludeProtocolDeserializers(): V1_MappingIncludeProtocolDeserializer[] {
+    return [
+      (json: PlainObject<V1_MappingInclude>): V1_MappingInclude | undefined => {
+        if (json._type === V1_MAPPING_INCLUDE_DATASPACE_TYPE) {
+          return V1_deserializeMappingInclude(json);
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  V1_getExtraMappingIncludeIdentifierBuilders(): V1_MappingIncludeIdentifierBuilder[] {
+    return [
+      (protocol: V1_MappingInclude): string | undefined => {
+        if (protocol instanceof V1_MappingIncludeDataSpace) {
+          return protocol.includedDataSpace;
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  override V1_getExtraExecutionInputCollectors(): V1_ExecutionInputCollector[] {
+    return [
+      (
+        graph: PureModel,
+        protocolGraph: V1_PureModelContextData,
+      ): V1_PackageableElement[] =>
+        protocolGraph.elements.filter(
+          (element) => element instanceof V1_DataSpace,
```

Comment:
dataspace relies on `Diagram` we we might also want to add that, also it could depend on `V1_Service` so we need to potentially add that here as well,

@MauricioUyaguari I'm worried about this particular change, should we just remove the `V1_getExtraExecutionInputCollectors` mechanism altogether at this point?

Reply:
Not following this ask...when I build this mapping all I need is the dataspace...which by pass two should already have been indexed right?
Is there something additional required when I attempt this resolve?

Reply:
Sorry I looked at the wrong diff...as of now is the claim that we would need to send more than the DS for execution?

Reply:
ok, don't worry about this change, we will need to do something on our end later, but it's somewhat a bigger change that's not fair to ask you to do.

Reply:
TODO: per discussion with @MauricioUyaguari we will completely remove `V1_getExtraExecutionInputCollectors` and make it so that we send the whole graph to engine for execution-based operations (need some slightly special handling for semi-interactive service registration flow)